02.calendar/calendar.rb

```diff
@@ -0,0 +1,25 @@
+require 'optparse'
+require 'date'
+
+options = ARGV.getopts('y:', 'm:')
+year = options['y']
+year ||= Date.today.year
+month = options['m']
+month ||= Date.today.month
+
+def calendar(year, month)
+  year = year.to_i
+  month = month.to_i
+  puts "      #{month}月 #{year}"
+  puts "日 月 火 水 木 金 土"
+  (1..Date.new(year, month, -1).day).each do |day|
+    date = Date.new(year, month, day)
```

Comment:
```suggestion
  (Date.new(year, month, 1)..Date.new(year, month, -1)).each do |date|
```

として、Dateオブジェクトのままループさせるとロジックもシンプルになっていいかもしれません。

Reply:
> Dateオブジェクトのままループさせるとロジックもシンプルになっていいかもしれません。

わー、ホントだ！気づかなかった…！！良い！ :pray:
なぜかループさせる時は数字にしなきゃいけないような気になっちゃってましたが、そもそも日付は日付のまま扱って回した方が話がシンプルになるんですね。

Reply:
[Loop as Date ofject · taea/ruby-practices@5333679](https://github.com/taea/ruby-practices/commit/5333679913e0abb77cdb046b739f33d40bf4b41e) にて対応